src/KafkaFlow.Retry.IntegrationTests/Core/Bootstrappers/BootstrapperKafka.cs

```diff
@@ -20,23 +22,32 @@ internal static class BootstrapperKafka
     {
         internal static async Task RecreateKafkaTopicsAsync(string kafkaBrokers, string[] topics)
         {
-            using (var adminClient = new Confluent.Kafka.AdminClientBuilder(new Confluent.Kafka.AdminClientConfig { BootstrapServers = kafkaBrokers }).Build())
+            using (var adminClient = new AdminClientBuilder(new AdminClientConfig { BootstrapServers = kafkaBrokers }).Build())
             {
                 foreach (var topic in topics)
                 {
-                    var topicMetadata = adminClient.GetMetadata(topic, TimeSpan.FromSeconds(20));
-                    if (topicMetadata.Topics.First().Partitions.Count > 0)
+                    var topicMetadata = adminClient
+                        .GetMetadata(topic, TimeSpan.FromSeconds(20))
+                        .Topics
+                        .First();
+
+                    if (topicMetadata.Error.Code == ErrorCode.UnknownTopicOrPart)
+                    {
+                        await adminClient.CreateTopicsAsync(new[] { new TopicSpecification { Name = topic } }).ConfigureAwait(false);
+                    }
+
+                    if (topicMetadata.Partitions.Count > 0)
                     {
                         try
                         {
-                            var deleteTopicRecords = new List<Confluent.Kafka.TopicPartitionOffset>();
-                            for (int i = 0; i < topicMetadata.Topics.First().Partitions.Count; i++)
+                            var deleteTopicRecords = new List<TopicPartitionOffset>();
```

Comment:
I think you can create the new `List<TopicPartitionOffset>(opicMetadata.Partitions.Count)`

Reply:
Yes! done.